ansible_base/lib/utils/auth.py

```diff
@@ -30,3 +35,18 @@ def get_team_model() -> Type[AbstractTeam]:
 
 def get_organization_model() -> Type[AbstractOrganization]:
     return get_model_from_settings('ANSIBLE_BASE_ORGANIZATION_MODEL')
+
+
+def get_object_by_ansible_id(qs: QuerySet, ansible_id: Union[str, UUID], annotate_as: str = 'ansible_id_for_filter') -> Model:
```

Comment:
We except this function to return 1 item. Because of that, do we need to feed this a `QuerySet` or could we just pass it in a `Model` object? If you named it cls you could get rid of line 43 and then on line 48 just do `return cls.objects.all().annotate...`

Reply:
I guess I was thinking that you would want to filter a queryset in related cases and get code reuse from that. But in that case you'd prefer a separate method for annotating, which would take the qs, and the get method could take the class. Although Django methods like get_object_or_404 tend to take a queryset as an argument.